src/rockstor/fs/btrfs.py

```diff
@@ -1002,8 +1010,39 @@ def balance_status(pool):
     return stats
 
 
-def device_scan():
-    return run_command([BTRFS, 'device', 'scan'])
+def device_scan(dev_byid_list=['all']):
+    """
+    When called with no parameters a 'btrfs device scan' is executed, ie a
+    system wide scan of all /dev block devices to update their btrfs status.
+    Otherwise the list of devices is iterated and a 'btrfs device scan dev'
+    is executed for each item in the passed list. Detached device names and
+    path names that don't exist are ignored.
+    :param dev_byid_list: list of byid device names (without paths) to perform
+    a 'btrfs device scan' on. If not supplied then a single element list
+    ['all'] is substituted and this flags a system wide scan request.
+    :return: (out, err, rc) of the first rc !=0 run or the last rc = 0 run.
+    """
+    out = err = ['']
+    # default to successful return code unless we find otherwise.
+    rc = 0
+    if len(dev_byid_list) > 0:
+        if dev_byid_list[0] == 'all':
+            return run_command([BTRFS, 'device', 'scan'])
+        for dev_byid in dev_byid_list:
+            if re.match('detached-', dev_byid) is not None:
+                # Skip detached devices as we know they don't exist.
+                # Potential log point for early detached device discovery.
+                continue
+            dev_byid_withpath = ('/dev/disk/by-id/%s' % dev_byid)
+            if os.path.exists(dev_byid_withpath):  # only scan existing devices
+                out, err, rc = run_command(
+                    [BTRFS, 'device', 'scan', dev_byid_withpath])
```

Comment:
even if dev_byid_withpath exists, this could potentially raise an exception. Since we are interested in the return code in both failure and success scenarios, it's best to use
 
run_command([BTRFS, 'device', 'scan', dev_byid_withpath], throw=False)

not sure if I am missing something here, but hope you'll agree with me otherwise.